src/app/components/dashboard-apps-dock/dashboard-apps-dock.coffee

```diff
@@ -0,0 +1,157 @@
+
+DashboardAppsDockCtrl = ($scope, $cookies, $uibModal, $window, MnoeOrganizations, MnoeAppInstances) ->
+  'ngInject'
+
+  $scope.appDock = {}
+  $scope.appDock.isMinimized = true
+  $scope.activeApp = null
+  $scope.launchApp = {isClosed: true}
+
+  # 'Lock' the dock when a menu is expanded.
+  # Ie: we disable all effects and animation
+  $scope.appDock.isAnimated = () ->
+    $scope.activeApp == null
+
+  # Scope initialization
+  $scope.modal ||= {}
+
+  # Mobile dock: max-height need to be an absolute value for the scrolling to work
+  pageHeight = jQuery(window).height()
+  navHeight = pageHeight * 0.7
+  jQuery('#mobile-dock').css({ maxHeight: navHeight + 'px' })
+
+  # ----------------------------------------------------------
+  # Permissions helper
+  # ----------------------------------------------------------
+  $scope.helper = {}
+  $scope.helper.displaySettings = ->
+    MnoeOrganizations.can.update.appInstance()
+
+  $scope.helper.dataSyncPath = (instance) ->
+    "/mnoe/webhook/oauth/#{instance.uid}/sync"
+
+  $scope.helper.isLaunchHidden = (app) ->
+    app.status == 'terminating' ||
+    app.status == 'terminated' ||
+    $scope.helper.isOauthConnectBtnShown(app) ||
+    $scope.helper.isNewOfficeApp(app)
+
+  $scope.helper.isOauthConnectBtnShown = (instance) ->
+    instance.app_nid != 'office-365' &&
+    instance.stack == 'connector' &&
+    !instance.oauth_keys_valid
+
+  $scope.helper.isNewOfficeApp = (instance) ->
+    instance.stack == 'connector' && instance.appNid == 'office-365' && (moment(instance.createdAt) > moment().subtract({minutes:5}))
+
+  $scope.helper.oAuthConnectPath = (instance)->
+    MnoeAppInstances.clearCache()
+    $window.location.href = "/mnoe/webhook/oauth/#{instance.uid}/authorize"
+          
+  # Create My Account Behavior
+  # If app requires custom we open the popup, otherwise we open the link directly
+  $scope.createAction = (app) ->
+    if app.customInfoRequired
+      $scope.showCustomInfo(app, 'create')
+    else
+      true # Don't cancel the link click
+
+    # Launch cloud application
+    # If app requires custom we open the popup, otherwise we open the link directly
+  $scope.launchAction = (app) ->
+    if app.customInfoRequired
+      $scope.showCustomInfo(app, 'launch')
+      return false
+    else
+      $window.open("/mnoe/launch/#{app.uid}", '_blank')
+      return true
+      
+  $scope.connectAction = (app) ->
+    if app.appNid == 'myob' || app.appNid == 'xero'
+      $scope.showCustomInfo(app, 'connect')
+    else
+    true # Don't cancel the link click
```

Comment:
Is this method used?

Also the else is useless, this is the javascript translation:
```
$scope.connectAction = function(app) {
  if (app.appNid === 'myob' || app.appNid === 'xero') {
    $scope.showCustomInfo(app, 'connect');
  } else {

  }
  return true;
};
```